example/src/main/java/io/netty/example/http/file/HttpStaticFileServerHandler.java

```diff
@@ -331,12 +330,33 @@ private static void sendError(ChannelHandlerContext ctx, HttpResponseStatus stat
      * @param ctx
      *            Context
      */
-    private static void sendNotModified(ChannelHandlerContext ctx) {
+    private static void sendNotModified(ChannelHandlerContext ctx, boolean keepAlive) {
         FullHttpResponse response = new DefaultFullHttpResponse(HTTP_1_1, NOT_MODIFIED);
         setDateHeader(response);
 
-        // Close the connection as soon as the error message is sent.
-        ctx.writeAndFlush(response).addListener(ChannelFutureListener.CLOSE);
+        sendAndCleanupConnection(ctx, response, keepAlive);
+    }
+
+    /**
+     * If Keep-Alive is disabled, attaches "Connection: close" header to the response
+     * and closes the connection after the response being sent.
+     */
+    private static void sendAndCleanupConnection(ChannelHandlerContext ctx, FullHttpResponse response,
+                                                 boolean keepAlive) {
+        HttpUtil.setContentLength(response, response.content().readableBytes());
+        if (!keepAlive) {
+            // We're going to close the connection as soon as the response is sent,
+            // so we should also make it clear for the client.
+            response.headers().set(HttpHeaderNames.CONNECTION, HttpHeaderValues.CLOSE);
+        }
+
+        ctx.write(response);
+        ChannelFuture lastWrite = ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT);
```

Comment:
@kachayev this is not correct as you already write a `FullHttpResonse` and so MUST NOT write an extra `LastHttpContent`